From 19caabbaeeff15eceba16067ac466a09979c6f7a Mon Sep 17 00:00:00 2001
From: jhenrique09 <jhenrique09.mcz@hotmail.com>
Date: Sun, 7 Mar 2021 01:58:18 -0300
Subject: [PATCH 30/34] FaceService: Nuke useless call to getAuthenticatorId

That value is static, so the call isn't needed. That fixes also a logspam:

2021-03-06 21:11:09.553 1767-1767/? W/Binder: Outgoing transactions from this process must be FLAG_ONEWAY
    java.lang.Throwable
        at android.os.BinderProxy.transact(BinderProxy.java:511)
        at com.android.internal.util.custom.faceunlock.IFaceService$Stub$Proxy.getAuthenticatorId(IFaceService.java:335)
        at com.android.server.biometrics.face.CustomFaceService.getAuthenticatorId(CustomFaceService.java:275)
        at com.android.server.biometrics.face.FaceService.updateActiveGroup(FaceService.java:1190)
        at com.android.server.biometrics.BiometricServiceBase.loadAuthenticatorIds(BiometricServiceBase.java:1183)
        at com.android.server.biometrics.face.CustomFaceService$FaceServiceConnection.lambda$onServiceConnected$0$CustomFaceService$FaceServiceConnection(CustomFaceService.java:427)
        at com.android.server.biometrics.face.-$$Lambda$CustomFaceService$FaceServiceConnection$xfTT3adGf0joHpCrnXRrbjd4jD8.run(Unknown Source:2)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at com.android.server.SystemServer.run(SystemServer.java:631)
        at com.android.server.SystemServer.main(SystemServer.java:417)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:925)

Change-Id: I98d5f4bad9af7faaea8d5333362d080f41a16c85
Signed-off-by: SamarV-121 <samarvispute121@gmail.com>
---
 .../com/android/server/biometrics/face/FaceService.java     | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/services/core/java/com/android/server/biometrics/face/FaceService.java b/services/core/java/com/android/server/biometrics/face/FaceService.java
index e43f4f9e..74363751 100644
--- a/services/core/java/com/android/server/biometrics/face/FaceService.java
+++ b/services/core/java/com/android/server/biometrics/face/FaceService.java
@@ -1185,11 +1185,7 @@ public class FaceService extends BiometricServiceBase {
             mCurrentUserId = userId;
             mCustomFaceService.setCurrentUserId(mCurrentUserId);
             if (mCustomFaceService.getService(mCurrentUserId) != null) {
-                long authId = 0;
-                if (hasEnrolledBiometrics(mCurrentUserId)) {
-                    authId = (long) mCustomFaceService.getAuthenticatorId();
-                }
-                mAuthenticatorIds.put(userId, authId);
+                mAuthenticatorIds.put(userId, (long) 1);
             } else {
                 mCustomFaceService.callForBind(userId);
                 Slog.w(TAG, "updateActiveGroup(): Face service not started!");
-- 
2.30.1

